sympy/polys/matrices/domainmatrix.py

```diff
@@ -234,3 +267,69 @@ def __eq__(A, B):
         if not isinstance(B, DomainMatrix):
             return NotImplemented
         return A.rep == B.rep
+
+
+class DomainScalar:
+    r"""
+    docstring
+    """
+
+    def __init__(self, element, domain):
+        self.element = element
+        self.domain = domain
+
+    @classmethod
+    def new(cls, element, domain):
+        return cls(element, domain)
+
+    def __repr__(self):
+        return repr(self.element)
+
+    @classmethod
+    def from_sympy(cls, expr):
+        [domain, [element]] = construct_domain([expr])
+        return cls.new(element, domain)
+
+    def to_domain(self, domain):
+        element = domain.convert_from(self.element, self.domain)
+        return self.new(element, domain)
+
+    def unify(self, other):
+        domain = self.domain.unify(other.domain)
+        return self.to_domain(domain), other.to_domain(domain)
+
+    def __add__(self, other):
+        if not isinstance(other, DomainScalar):
+            return NotImplemented
+        self, other = self.unify(other)
+        return self.new(self.element + other.element, self.domain)
+
+    def __sub__(self, other):
+        if not isinstance(other, DomainScalar):
+            return NotImplemented
+        self, other = self.unify(other)
+        return self.new(self.element - other.element, self.domain)
+
+    def __mul__(self, other):
+        if not isinstance(other, DomainScalar):
+            return NotImplemented
+        self, other = self.unify(other)
+        return self.new(self.element * other.element, self.domain)
+
+    def __floordiv__(self, other):
+        if not isinstance(other, DomainScalar):
+            return NotImplemented
+        return self.from_sympy(self.element // other.element)
+
+    def __pow__(self, n):
+        if self.element == 0 and n == -1:
+            raise ZeroDivisionError
+        return self.from_sympy(self.element**n)
```

Comment:
Why is this using `from_sympy`?

Is `self.element ** n` not a domain element?

This should check that `n` is an `int` and otherwise return `NotImplemented`.

What if `n == -2` and element is zero?

Reply:
> Why is this using `from_sympy`?
> 
Sorry, I missed that it will already be a domain element

> This should check that `n` is an `int` and otherwise return `NotImplemented`.
> 
Sure, will do so.

> What if `n == -2` and element is zero?
I missed it. My bad.

Thanks.